Source/SmallBasic.Editor/Components/Pages/Edit/LibraryExplorer.cs

```diff
@@ -107,8 +107,44 @@ protected override void ComposeTree(TreeComposer composer)
                                             composer.Element(
                                                 name: library.Name.Length > 10 ? "long-name" : "short-name",
                                                 body: () => composer.Text(library.Name));
+
+                                            bool hasFunctioningParts = false;
```

Comment:
I suggest replacing these with something like:

`bool hasNonFunctioningParts = library.Properties.Any(p => p.Check) || library.Methods.Any(m => m.Check);`

This is (1) more concise, and (2) stops iterating as soon as a match is found.